src/server/pfs/server/driver_new_storage.go

```diff
@@ -97,54 +104,43 @@ func (d *driver) getFilesNewStorageLayer(ctx context.Context, repo, commit, glob
 	return mr.Get(w)
 }
 
-func (d *driver) compact(ctx context.Context, outputPath string, prefixes []string) error {
-	// (bryce) need some cleanup improvements, probably garbage collection.
-	if err := d.storage.Delete(ctx, tmpPrefix); err != nil {
-		return err
-	}
-	// (bryce) need to add a resiliency measure for existing incomplete compaction for the prefix (master crashed).
-	// Setup task.
-	task := &work.Task{Id: prefixes[0]}
-	var err error
-	task.Data, err = serializeCompaction(&pfs.Compaction{Prefixes: prefixes})
-	if err != nil {
-		return err
-	}
-	if err := d.storage.Shard(ctx, prefixes, func(pathRange *index.PathRange) error {
+func (d *driver) compact(master *work.Master, outputPath string, inputPrefixes []string) (retErr error) {
+	scratch := path.Join(tmpPrefix, uuid.NewWithoutDashes())
+	defer func() {
+		// (bryce) need some cleanup improvements, probably garbage collection.
+		if err := d.storage.Delete(context.Background(), scratch); retErr == nil {
```

Comment:
`context.Background` here as well, should probably be a more meaningful context.

Reply:
We still want the cleanup code to run if the context is cancelled. But, this again will be handled by garbage collection, so I don't think it is particularly important for getting this PR in (a lot of this integration work in general is going to require some temporary hacks so that we can get this stuff in digestible pieces, but I do make notes of all these places as you can see).